app/src/main/java/com/globant/weatherapp/mvp/view/WeatherView.kt

```diff
@@ -3,18 +3,24 @@ package com.globant.weatherapp.mvp.view
 import android.app.Activity
 import android.view.View
 import android.widget.Toast
+import androidx.fragment.app.FragmentActivity
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.globant.weatherapp.R
-import com.globant.weatherapp.data.entities.FiveDaysWeather
-import com.globant.weatherapp.mvp.contracts.WeatherContracts
+import com.globant.weatherapp.activity.WeatherDetailFragment
+import com.globant.weatherapp.data.entity.FiveDaysWeather
+import com.globant.weatherapp.mvp.contract.WeatherContract
 import com.globant.weatherapp.mvp.view.base.ActivityView
-import com.globant.weatherapp.utils.WeatherRecyclerViewAdapter
+import com.globant.weatherapp.util.WeatherRecyclerViewAdapter
 import kotlinx.android.synthetic.main.activity_weather_info_layout.recycler_view
 import kotlinx.android.synthetic.main.activity_weather_info_layout.progress_bar
 
-class WeatherView(activity: Activity) : ActivityView(activity), WeatherContracts.View {
+class WeatherView(activity: Activity) : ActivityView(activity), WeatherContract.View {
 
-    private var weatherAdapter: WeatherRecyclerViewAdapter = WeatherRecyclerViewAdapter()
+    private var weatherAdapter: WeatherRecyclerViewAdapter =
+        WeatherRecyclerViewAdapter { date: String, cityId: Int ->
+            val fragment = WeatherDetailFragment.newInstance(cityId, date)
```

Comment:
this val is necessary?

Reply:
Don't. Refactored, thanks Nico, I didnt know that I could use the show method without an instance of fragment.